### PR TITLE
feat: included rounds of questions

### PR DESCRIPTION
**Pull request by Nishanthh**

### CODE REVIEW

File: src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts [hunk 1]
```diff
@@ -77,4 +82,22 @@ export class ConsultationPublicResponsesComponent implements OnInit, AfterViewCh
     });
   }
 
+  getActiveRound(profileData) {
+    if (profileData && profileData.responseRounds) {
+      const responseRounds = profileData.responseRounds;
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts:87]: @Nishanthh I don't think this is necessary. You already have  responseRounds defined in line 44

File: src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts [hunk 2]
```diff
@@ -37,8 +40,10 @@ export class ConsultationPublicResponsesComponent implements OnInit, AfterViewCh
     this.consultationService.consultationProfileData.subscribe((data) => {
       if (data) {
         this.profileData = data;
-        this.responseQuestions = data.questions;
+        this.roundNumber = this.getActiveRound(data);
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts:43]: @Nishanthh Would be more clear if you rename this to `activeRoundNumber`

File: src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts [hunk 3]
```diff
@@ -77,4 +82,22 @@ export class ConsultationPublicResponsesComponent implements OnInit, AfterViewCh
...
+        const activeRound  = responseRounds.find((round) => round.active);
+        if (!isObjectEmpty(activeRound)) {
+          return activeRound.roundNumber;
+        }
+      }
+    }
+    return;
+  }
+
+  setActiveRound(roundNumber) {
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts:98]: @Nishanthh Is this being used? I can see line 46 already doing what this function is supposed to do

File: src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts [hunk 4]
```diff
@@ -13,11 +14,13 @@ export class ConsultationPublicResponsesComponent implements OnInit, AfterViewCh
 
   profileData: any;
   responseList: any;
-  responseQuestions: any;
```
Comment (avikt) [inline src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts:16]: @Nishanthh This shouldn't be deleted, right? We will still need to display the response questions in the UI

File: src/app/modules/consultations/consultations-summary/consultations-summary.component.ts [hunk 5]
```diff
@@ -91,11 +99,32 @@ export class ConsultationsSummaryComponent implements OnInit {
...
       this.summaryData = {
         publicResponseCount : this.publicResponses.length,
         title: 'RESPONSES TO'
       };
     }
   }
 
+  getActiveRound(profileData) {
+    if (profileData && profileData.responseRounds) {
+      const responseRounds = profileData.responseRounds;
```
Comment (avikt) [inline src/app/modules/consultations/consultations-summary/consultations-summary.component.ts:113]: @nishanthh Again, I don't think you need to define a function variable for responseRounds as there is already defined in line 84

File: src/app/modules/consultations/consultations-summary/consultations-summary.component.ts [hunk 6]
```diff
@@ -91,11 +99,32 @@ export class ConsultationsSummaryComponent implements OnInit {
...
+        const activeRound  = responseRounds.find((round) => round.active);
+        if (!isObjectEmpty(activeRound)) {
+          return activeRound.roundNumber;
+        }
+      }
+    }
+    return;
+  }
+
+  setActiveRound(roundNumber) {
```
Comment (avikt) [inline src/app/modules/consultations/consultations-summary/consultations-summary.component.ts:124]: @nishanthh Do we need this function? I don't see it being called anywhere

File: src/app/shared/components/response-answers/response-answers.component.ts [hunk 7]
```diff
@@ -7,64 +9,86 @@ import { Component, OnInit, Input, ChangeDetectorRef } from '@angular/core';
...
+        });
+        return responseAnswers;
+      }
     }
     return;
   }
 
+  getQuestions() {
+    let questions;
+    if (this.showOnlyLongTextAnswer) {
```
Comment (avikt) [inline src/app/shared/components/response-answers/response-answers.component.ts:79]: @Nishanthh Here both the if and the else blocks have the same code running. Was something missed?

File: src/app/shared/components/response-answers/response-answers.component.ts [hunk 8]
```diff
@@ -7,64 +9,86 @@ import { Component, OnInit, Input, ChangeDetectorRef } from '@angular/core';
...
+       questions =
+        this.consultationService.getQuestions(this.response.consultation, this.response.roundNumber);
+        return questions;
+    } else {
+      if (this.roundNumber) {
+        questions = this.consultationService.getQuestions(this.response.consultation, this.roundNumber);
+        return questions;
+      }
+      return this.consultationService.getQuestions(this.response.consultation);
```
Comment (avikt) [inline src/app/shared/components/response-answers/response-answers.component.ts:88]: @Nishanthh Why is this return statement there? Is it necessary?

**Review by avikt — CHANGES_REQUESTED** [review]
@Nishanthh Need some clarifications on a few points and left feedback

File: src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts (hunk 3 above)
Comment (Nishanthh) [reply src/app/modules/consultations/consultation-profile/consultation-public-responses/consultation-public-responses.component.ts:100]: @avikt in changing the round, to update we round number we need this function bro.

File: src/app/shared/components/response-answers/response-answers.component.ts (hunk 7 above)
Comment (Nishanthh) [reply src/app/shared/components/response-answers/response-answers.component.ts:79]: @avikt No Nothing missed here, its a old flow when no tabs in public responses, now modified

File: src/app/shared/components/response-answers/response-answers.component.ts (hunk 8 above)
Comment (Nishanthh) [reply src/app/shared/components/response-answers/response-answers.component.ts:88]: @avikt needed modified it

**Review by avikt — APPROVED** [review]

**Pull request merged**